src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -55,6 +55,18 @@ const TopicFragment = ({
       following: item.following !== undefined ? item.following : item.user_id_follower !== null
     }));
   }, [discovery.initialTopics]);
+  const newMapFollowedTopics = React.useMemo(() => {
+    return followedTopic.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [followedTopic]);
+  const newMapUnfollowedTopics = React.useMemo(() => {
+    return unfollowedTopic.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [unfollowedTopic]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useMemo` hooks for `newMapFollowedTopics` and `newMapUnfollowedTopics` are mapping over the `followedTopic` and `unfollowedTopic` arrays respectively. However, they are not doing any expensive computations that would benefit from memoization. Instead, they are creating a new array with the same structure as the original one. This is unnecessary and could be simplified by directly using the original arrays.
```diff
-  const newMapFollowedTopics = React.useMemo(() => {
-    return followedTopic.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [followedTopic]);
-  const newMapUnfollowedTopics = React.useMemo(() => {
-    return unfollowedTopic.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
-  }, [unfollowedTopic]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->